scipy/optimize/_direct_py.py

```diff
@@ -0,0 +1,174 @@
+# -*- coding: utf-8 -*-
+r"""
+scipydirect - A python wrapper to the DIRECT algorithm.
+=======================================================
+
+DIRECT is a method to solve global bound constraint optimization problems and
+was originally developed by D. R. Jones, C. D. Perttunen and B. E. Stuckmann.
+It is designed to find **global** solutions of mathematical optimization problems of the from
+
+.. math::
+
+       \min_ {x \in R^n} f(x)
+
+subject to
+
+.. math::
+
+       x_L \leq  x  \leq x_U
+
+Where :math:`x` are the optimization variables (with upper and lower
+bounds), :math:`f(x)` is the objective function.
+
+The DIRECT package uses the Fortran implementation of DIRECT written by
+Joerg.M.Gablonsky, DIRECT Version 2.0.4. More information on the DIRECT
+algorithm can be found in Gablonsky's `thesis <http://repository.lib.ncsu.edu/ir/bitstream/1840.16/3920/1/etd.pdf>`_.
+
+.. codeauthor:: Andreas Mayer <andimscience@gmail.com>, Amit Aides <amitibo@tx.technion.ac.il>
+"""
+
+from __future__ import print_function
+import numpy as np
+from ._direct import direct
+from .optimize import OptimizeResult
+
+ERROR_MESSAGES = (
+    'u[i] < l[i] for some i',
+    'maxfun is too large',
+    'Initialization failed',
+    'There was an error in the creation of the sample points',
+    'An error occured while the function was sampled',
+    'Maximum number of levels has been reached.',
+)
+
+SUCCESS_MESSAGES = (
+    'Number of function evaluations done is larger then maxfun',
+    'Number of iterations is equal to maxiter',
+    'The best function value found is within fglper of the (known) global optimum',
+    'The volume of the hyperrectangle with best function value found is below volper',
+    'The volume of the hyperrectangle with best function value found is smaller then volper'
+)
+
+def _minimize_direct(func, bounds=None, nvar=None, args=(), disp=False,
+                     eps=1e-4, maxfun=20000, maxiter=6000, method=0,
+                     fglobal=-1e100, fglper=0.01, volper=-1.0, sigmaper=-1.0):
+    r"""
+
+    Solve an optimization problem using the DIRECT (Dividing Rectangles) algorithm.
+    It can be used to solve general nonlinear programming problems of the form:
+
+    .. math::
+
+           \min_ {x \in R^n} f(x)
+
+    subject to
+
+    .. math::
+
+           x_L \leq  x  \leq x_U
+    
+    Where :math:`x` are the optimization variables (with upper and lower
+    bounds), :math:`f(x)` is the objective function.
+
+    Parameters
+    ----------
+    func : objective function
+        called as `func(x, *args)`; does not need to be defined everywhere,
+        raise an Exception where function is not defined
+    bounds : array_like, optional
+            ``(min, max)`` pairs for each element in ``x``, defining
+            the bounds on that parameter.
+    nvar : int, optional
+        Dimensionality of x (only needed if `bounds` is not defined)
+    eps : float, optional
```

Comment:
The naming of all the tolerances need to be rethought. Does the minimisation stop if the change in function value between iterations is less than `eps`? If so, then this parameter should be called `fatol`.

Reply:
I am not sure if that is what the meaning of `eps` is, because, `fglper` does the same task though the algorithm steps when the percentage error satisfies a certain function. So, may be `eps` is something different which is used during the algorithm.

Reply:
In [this document](http://www2.peq.coppe.ufrj.br/Pessoal/Professores/Arge/COQ897/Naturais/DirectUserGuide.pdf), definition 3.2 on page 7 clarifies the meaning of `eps` here. I think that `eps` and `fatol` are different.

Reply:
Found something that might be helpful in general here: 
[Jablonsky's thesis](https://repository.lib.ncsu.edu/bitstream/handle/1840.16/3920/etd.pdf?sequence=1) contains a description of all parameters. Epsilon is explained in sec. 3.3. It also contains a nice description of the Fortran code we are using here.